### PR TITLE
nucleotide_base complement table static member functions

### PR DESCRIPTION
**Pull request by marehr**
Part of https://github.com/seqan/product_backlog/issues/295

### CODE REVIEW

File: CHANGELOG.md
Comment (eseiler) [inline CHANGELOG.md:133]:
```suggestion
* We relaxed some requirements of `seqan3::alphabet_base<alphabet_t>`
  ([\#2427](https://github.com/seqan/seqan3/pull/2427)):
  * Old requirements: `alphabet_t::rank_to_char` and `alphabet_t::char_to_rank` must be lookup tables.
  * New requirements: `alphabet_t::rank_to_char` and `alphabet_t::char_to_rank` must be static member functions.

  This allows for more flexible rank <-> char conversion implementations. Lookup tables are still possible within those 
  static member functions. However, alphabets that do not need a lookup table can now use easier and/or more efficient
  implementations. For example, `seqan3::gap` always returns rank `0` or char `-`, or `seqan3::phred42` where the rank 
  and char representations are offset by a fixed value.
* We relaxed a requirement of `seqan3::nucleotide_base<alphabet_t>`
  ([\#2584](https://github.com/seqan/seqan3/pull/2584)):
  * Old requirement: `alphabet_t::complement_table` must be a lookup table.
  * New requirement: `alphabet_t::rank_complement` must be a static member function.
```

**Review by eseiler — CHANGES_REQUESTED** [review]
Just changelog re-write suggestion

Edit: I had this tab open when I did reviews, so you get the second review too early... :)

**Review by eseiler — APPROVED** [review]

**Review by SGSSGene — APPROVED** [review]

**Pull request merged**
